.github/workflows/nightly.yml

```diff
@@ -0,0 +1,104 @@
+name: Publish Nightly Builds
+
+on:
+  push:
+    branches:
+      - nightly_build
+env:
+  JAVA_DISTRIBUTION: 'corretto'
+  JAVA_VERSION: '11'
+jobs:
+  short-hash-getter:
+    runs-on: ubuntu-latest
+    outputs:
+      short-hash: ${{ steps.getter.outputs.SHORT_HASH }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Get Short Hash
+        id: getter
+        run: |
+          echo "SHORT_HASH=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+  build-test-publish:
+    needs: [short-hash-getter] # needed to get the short hash
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - runson: macos-14 # mac arm
+            extraarch: "{for_setup_java: 'x64', for_env: 'X64'}" # we test x64 also using rosetta
+            archive: async-profiler-3.0-macos.zip
```

Comment:
Can we avoid hard-coding version number? It may change frequently in the future.

Reply:
On the contrary, parametrizing `extraarch` is not necessary: it will be only x64 and only on macOS.